megamek/src/megamek/common/templates/AeroTROView.java

```diff
@@ -172,14 +178,50 @@ protected int addWeaponBays(String[][] arcSets) {
         return nameWidth;
     }
 
+    
+    /**
+     * Computes any modification to the aerospace AV for linked Artemis, Apollo, or PPC capacitor
+     *
+     * @param weapon   The type of weapon
+     * @param linkedBy The type of equipment the weapon is linked by
+     * @param bay      Whether the weapon is part of a weapon bay
+     * @return         The AV modification, if any
+     */
+    private int aeroAVMod(WeaponType weapon, EquipmentType linkedBy, boolean bay) {
+        if (linkedBy.hasFlag(MiscType.F_ARTEMIS)
+                || linkedBy.hasFlag(MiscType.F_ARTEMIS_V)) {
+            // The 9 and 10 rows of the cluster hits table is only different in the 3 column
+            if (weapon.getAtClass() == WeaponType.CLASS_MML) {
+                if (weapon.getRackSize() >= 7) {
+                    return 2;
+                } else if ((weapon.getRackSize() >= 5) || linkedBy.hasFlag(MiscType.F_ARTEMIS_V)) {
+                    return 1;
+                }
+            } else if (weapon instanceof LRMWeapon) {
+                return weapon.getRackSize() / 5;
+            } else if (weapon instanceof SRMWeapon) {
+                return 2;
+            }
+        } else if (linkedBy.hasFlag(MiscType.F_ARTEMIS_PROTO) && weapon.getRackSize() == 2) {
+            // The +1 cluster hit bonus only adds a missile hit for SRM2
+            return 2;
+        } else if (bay && linkedBy.hasFlag(MiscType.F_PPC_CAPACITOR)) {
+            // PPC capacitors in weapon bays are treated as if always charged
+            return 5;
+        }
+        return 0;
```

Comment:
Since you are going to have to fix all those implicit castings to get through CodeQL, can you see if you can do something about those magic numbers? I dont know if a builder, a data class, something that is a little less cryptic.
I know thats how 97% of the code is set, but I'm only asking for it if you can find a reasonable solution other than creating a bunch of `static final int RACKSIZE_2 = 2;` and the likes of it.
Maybe moving it to a class at least, so it is self contained like... `AeroAVModCalculator.getMod(...)`, later we could extract an interface or abstract class and replicate it in multiple places. (Don't create the abstract/interface before it is needed)